src/repositories/soft-crud.repository.base.ts

```diff
@@ -175,8 +175,15 @@ export abstract class SoftCrudRepository<
         [pk]: id,
       } as Condition<T>;
     }
+    if (filter.fields) {
+      filter.fields = {
+        ...filter.fields,
+        deleted: true,
+      };
+    }
     const entity = await super.findById(id, filter, options);
     if (entity && !entity.deleted) {
+      delete entity.deleted;
```

Comment:
why removing this everytime ? What if someone was notnusing fields filter ?

Reply:
Fixed